### PR TITLE
Add width and height properties to images

### PR DESCRIPTION
**Pull request by zackkrida**
## Fixes

Related to #1034 by @zackkrida 

## Description

Adds width and height properties to images, as well as setting the background to grey while they are loading. We _could_ and maybe _should_ wait to display images until they are loaded, and show the skeleton components while they are loading, but this still makes an improvement to the current loading appearance via the grey backgrounds. 

## Testing Instructions

Observe the appearance of the all results grid and image results views while loading.

> before

CleanShot 2022-03-02 at 18 46 14@2x

CleanShot 2022-03-02 at 18 46 35@2x

> after (loading appearance while on a throttled 4g connection)

CleanShot 2022-03-02 at 19 21 26@2x

CleanShot 2022-03-02 at 19 19 11@2x

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: src/components/VAllResultsGrid/VImageCellSquare.vue [hunk 1]
```diff
@@ -13,12 +13,12 @@
     >
       <img
         ref="img"
-        class="w-full h-full object-cover rounded-sm"
+        class="w-full h-full object-cover rounded-sm bg-dark-charcoal-10 text-dark-charcoal-10"
         loading="lazy"
         :alt="image.title"
         :src="getImageUrl(image)"
-        :width="image.width"
-        :height="image.height"
```
Comment (zackkrida) [inline src/components/VAllResultsGrid/VImageCellSquare.vue:21]: these properties aren't actually defined, so it's better to use a default with the correct aspect ratio, for now.

Comment (obulat) [reply src/components/VAllResultsGrid/VImageCellSquare.vue:21]: Are these properties never defined and will never be used? Or could we have something like `"image.width ?? 250"`?

**Review by obulat — APPROVED** [review]
This creates a much nicer experience!

Could we also maybe add skeleton loaders for when we click on 'Load more', before the images are loaded? Now it takes some time for the gray squares to appear, and then some more time for the images to show.

File: src/components/VAllResultsGrid/VImageCellSquare.vue (hunk 1 above)
Comment (dhruvkb) [reply src/components/VAllResultsGrid/VImageCellSquare.vue:21]: I second @obulat's comment, that'll allow us to send the dimensions from the backend without needed a frontend change/deploy.

Comment (zackkrida) [reply src/components/VAllResultsGrid/VImageCellSquare.vue:21]: > absolutely should have dimensions coming from the API.

Yes, but none of our provider scripts set these fields. This suggestion is well-taken though! I'll do it this way.

**Review by sarayourfriend — APPROVED** [review]
It's not perfect (especially for the images-only search) but definitely an improvement! Getting the dimensions into the catalog would do wonders to fix this problem altogether.

File: src/components/VAllResultsGrid/VImageCellSquare.vue (hunk 1 above)
Comment (krysal) [reply src/components/VAllResultsGrid/VImageCellSquare.vue:21]: In fact, the StockSnap and the WordPress Photos Directory scripts do assign these values, for the rest it is a matter of updating them.

https://github.com/WordPress/openverse-catalog/blob/main/openverse_catalog/dags/providers/provider_api_scripts/stocksnap.py#L141-L142

https://github.com/WordPress/openverse-catalog/blob/main/openverse_catalog/dags/providers/provider_api_scripts/wordpress.py#L259-L260

Comment (zackkrida) [reply src/components/VAllResultsGrid/VImageCellSquare.vue:21]: That's great! Unfortunately those properties aren't exposed on the list view, just in the single result view:

https://api.openverse.engineering/v1/images/?q=dog&source=stocksnap

**Pull request merged**